feature-pitches/002-file-based-workflows/002-file-based-workflows.md

```diff
@@ -0,0 +1,106 @@
+
+### Improving workflow user awareness: A workflow file template
+
+In order to increase workflow usage, we also need to make users aware of them. To this end, it is worth considering adding a template workflow file to the default Renku project structure. For example, this could be a workflow file with a few example workflow steps written out but commented out. The user can uncomment it and replace the placeholders with their command, inputs, output, etc. There could also be a comment at the top of the file describing how to run the workflow.
+
+This would put workflows in front of more users, increasing awareness and hopefully also adoption. 
+
+### UX Note: Re-runnability & Pre-existing Outputs
+
+A common frustration users have with current Renku workflows is that the `workflow` & `run` (mostly `run`) commands errors when you run a workflow command that writes an output file that already exists.
+
+Users do this frequently during development: run script `a`, generate file `b`. Change script `a`, rerun, generate a new b. Great, now you're happy with script `a`, so `renku run script_a` one last time to record it. But now Renku complains that no outputs were generated (because file `b` already existed, so nothing was detected)! Users *can* avoid this by telling Renku explicitly that the output is file `b`, but that's more typing. More commonly, users end up deleting output files before every `renku run`, which they- very reasonably- don't like doing.
+
+The file-based workflow feature should solve this. The workflow should read the specified output in the workflow file, rather than detecting changed files. This removes the need for the user to delete the file so that Renku can detect its creation.
+
+
+## 🐰 Rabbit Holes
+
+### Venturing into the Metadata Realm
+
+This pitch will require a substantial effort to make the metadata all play nice. Without diving into implementation details, here are a few high-level requirments that the metadata must meet in order to provide a good user experience:
+
+* The workflows defined in files should be fully represented in the Renku KG metadata.
+* However, the workflow file itself should be the single source of truth for the workflow definition (not the KG metadata).
+* File based workflows can only be edited via the file. These workflows cannot be edited via the CLI or (in the future) somewhere in the RenkuLab UI.
```

Comment:
why not in the UI? That could also just modify the file under the hood.

Reply:
Yes the file can be edited through the UI as well, I think, but not the metadata directly, in isolation.

Something like `The metadata of file-based workflows can only be edited simultaneously with the file, not separately`

Reply:
I guess the two could be linked? I.e. I could do 

```
renku workflow edit --description "bla" my-workflow
```

and the command would edit the workflow in the metadata _and_ in the file?

Reply:
Unless this is super easy to implement, I would leave this out of scope, because I don't imagine many users would care about being able to do this. And rather return am error message that file-based workflows can only be edited via the file.

Reply:
sure that's fine - lets add it to the out-of-scope section